src/tqec/detectors/operation.py

```diff
@@ -0,0 +1,244 @@
+        self._local_coordinate_system_origin = new_origin
+
+
+class Detector(RelativeMeasurementsRecord):
+    def __init__(
+        self,
+        local_coordinate_system_origin: cirq.GridQubit,
+        data: list[RelativeMeasurementData],
+        time_coordinate: int = 0,
+    ) -> None:
+        """Operation representing a detector.
+
+        :param local_coordinate_system_origin: origin of the local coordinate system.
+            The origin along with the local coordinate system will be pinned to the
+            global coordinate system to resolve the actual qubit coordinates the
+            measurements applied to.
+        :param data: a list of :class:`RelativeMeasurementData` that composed the
+            relative measurements' operation.
+        :param time_coordinate: an annotation that will be forwarded to the DETECTOR
+            Stim structure as the last coordinate.
+        """
+        super().__init__(local_coordinate_system_origin, data)
+        self._time_coordinate = time_coordinate
+
+    def _circuit_diagram_info_(self, _: Any) -> str:
+        row, col, t = self.coordinates
+        return f"Detector({row},{col},{t})"
+
+    @property
+    def coordinates(self) -> tuple[int, ...]:
+        return (
+            self._local_coordinate_system_origin.row,
+            self._local_coordinate_system_origin.col,
+            self._time_coordinate,
+        )
+
+
+class Observable(RelativeMeasurementsRecord):
+    def __init__(
+        self,
+        local_coordinate_system_origin: cirq.GridQubit,
+        data: list[RelativeMeasurementData],
+        observable_index: int = 0,
+    ) -> None:
+        """Operation representing an observable.
+
+        :param local_coordinate_system_origin: origin of the local coordinate system.
+            The origin along with the local coordinate system will be pinned to the
+            global coordinate system to resolve the actual qubit coordinates the
+            measurements applied to.
+        :param data: a list of :class:`RelativeMeasurementData` that composed the
+            relative measurements' operation.
+        :param observable_index: the index of the observable.
+        """
+        super().__init__(local_coordinate_system_origin, data)
+        self._observable_index = observable_index
+
+    def _circuit_diagram_info_(self, _: cirq.CircuitDiagramInfoArgs):
+        return f"Observable({self._observable_index})"
+
+    @property
+    def index(self) -> int:
+        """Return the index of the observable."""
+        return self._observable_index
+
+
+def make_shift_coords(*shift: int) -> cirq.Operation:
+    """This is a helper function to make a :class:~ShiftCoords operation with the
+    `cirq.VirtualTag` tag.
+
+    :param shift: How much to shift each coordinate.
+
+    :return: A :class:`ShiftCoords` operation with the `cirq.VirtualTag` tag.
+    """
+    return ShiftCoords(*shift).with_tags(cirq.VirtualTag(), STIM_TAG)
```

Comment:
```suggestion
def make_shift_coords(*shifts: int) -> cirq.Operation:
    """This is a helper function to make a :class:~ShiftCoords operation with the
    `cirq.VirtualTag` tag.

    :param shifts: How much to shift each coordinate.

    :return: A :class:`ShiftCoords` operation with the `cirq.VirtualTag` tag.
    """
    return ShiftCoords(*shifts).with_tags(cirq.VirtualTag(), STIM_TAG)
```
see previous comment.